src/components/wizards/AddTestWizard/AddTestWizard.tsx

```diff
@@ -0,0 +1,293 @@
+      form.setFieldsValue({
+        file: null,
+      });
+
+      form.validateFields(['file']);
+    } else {
+      const readFile = new FileReader();
+
+      readFile.onload = e => {
+        if (e && e.target) {
+          const fileContent = e.target.result;
+
+          if (fileContent) {
+            form.setFieldsValue({
+              file: {
+                fileContent: fileContent as string,
+                fileName: file.file.name,
+              },
+            });
+
+            form.validateFields(['file']);
+          }
+        }
+      };
+
+      // @ts-ignore
+      readFile.readAsText(file.file);
+    }
+  };
+
+  const renderFormItems = (array: any) => {
+    return array.map((formItem: any) => {
+      const {
+        options = null,
+        itemLabel,
+        tooltip,
+        fieldName,
+        inputType,
+        modificator = null,
+        rules = [],
+        help = null,
+      } = formItem;
+
+      let children: Nullable<ReactElement<any, any>> = null;
+
+      if (inputType === 'select') {
+        children = (
+          <Select>
+            {options.map((option: any) => {
+              return (
+                <Option value={option.value} key={option.value}>
+                  {option.label}
+                </Option>
+              );
+            })}
+          </Select>
+        );
+      }
+
+      if (inputType === 'radio') {
+        children = (
+          <Radio.Group>
+            {options.map((option: any) => {
+              return (
+                <Radio value={option.value} key={option.value}>
+                  {option.label}
+                </Radio>
+              );
+            })}
+          </Radio.Group>
+        );
+      }
+
+      if (inputType === 'default') {
+        if (modificator) {
+          if (modificator === 'password') {
+            children = <Input.Password />;
+          }
+        } else {
+          children = <Input />;
+        }
+      }
+
+      if (inputType === 'uploadWithInput') {
+        children = <UploadWithInput onFileChange={onFileChange} />;
+      }
+
+      if (inputType === 'textarea') {
+        children = <TextArea rows={10} />;
+      }
+
+      if (!children) {
+        return null;
+      }
+
+      return (
+        <FormItem rules={rules} label={itemLabel} name={fieldName} tooltip={tooltip} key={fieldName} help={help}>
+          {children}
+        </FormItem>
+      );
+    });
+  };
+
+  return (
+    <StyledWizardContainer>
+      <StyledWizardTitle>{wizardTitle}</StyledWizardTitle>
+      <StyledWizardBody>
+        <StyledWizardForm>
+          <span>Create a new test based on file content, string or git based data.</span>
+          <Form
+            form={form}
+            name="add-test-form"
+            labelCol={{span: 8}}
+            onFinish={onFinish}
+            style={{paddingTop: 30}}
+            initialValues={{testSource: '', testType: '', branch: 'main'}}
+            onFieldsChange={() => {
```

Comment:
Maybe better to create new function and move this code there to improve readability